internal/config/envconfig/load.go

```diff
@@ -0,0 +1,11 @@
+package envconfig
+
+import (
+	"github.com/git-town/git-town/v19/internal/config/configdomain"
+)
+
+func Load() configdomain.PartialConfig {
```

Comment:
Just wanted to say that it's a brilliant approach to return a `PartialConfig` instance that contains all the data from env variables here and treat the environment like another source for config data!

This opens up the possibility to set Git Town settings via env variables in the future, if the need for that ever arises!

Reply:
Yeah the existing architecture allowed me to extend this nicely without touching the other configs :-). I really like how the separation of concern is handled in this repo. The boundaries of the domains are defined super clearly/strictly ❤️.

One Q I have in mind is should the env config takes precedence over everything else? I mean if usually we would override things for the repository with local config right ? 

Should we just inverse the priority ?

Reply:
> Yeah the existing architecture allowed me to extend this nicely without touching the other configs :-). I really like how the separation of concern is handled in this repo. The boundaries of the domains are defined super clearly/strictly ❤️.

Thanks for recognizing this! It took _a lot_ of refactoring to identify the various domains and concerns and model them orthogonal to each other. I think that's time well spent, because it makes maintaining this codebase doable, predictable, and fun!

Reply:
> One Q I have in mind is should the env config takes precedence over everything else? I mean if usually we would override things for the repository with local config right ?

That's a good question. I think the env config applies to all Git repos and multiple apps (the GitHub CLI and Git Town both use `GITHUB_TOKEN`), but the Git config applies only to Git Town and sometimes only to a single repo. Hence the Git config should override the env config, because more specific overrides more generic.

The env config should override the config file, though. Because the config file applies to all users, while the env config applies only to one user.

Reply:
> Hence the Git config should override the env config, because more specific overrides more generic.

Hmm, I disagree. Environment variables can be as specific as per-command, and we can't tell where the environment variable comes from. I think if anything is set in the environment, it should override everything else.

For example, running `GIT_CONFIG_PARAMETERS="'color.ui=never'" git status` will show status output without color, even if the Git config for this repo has `color.ui = always`.

Reply:
Cool. I think the current behaviour is fine then 👍

Reply:
Yes, you were right all along! It just took me a while to reach your level of insight :smile:

Reply:
fixed with this [commit](https://github.com/git-town/git-town/pull/4728/commits/abaeda0af3379249dacbd82b7e7bac879c31ab87). Had to instantiate Lineage in the `EmptyPartialConfig` to avoid `panic: assignment to entry in nil map`